### ISSUE TITLE
gettext builder fails on one file, when trying to process StringList as a path

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#915]
sphinx-build using gettext builder makes it through a good portion of the project and then fails with the following error:

```
writing message catalogs... [ 20&#37;] reference
Exception occurred:
  File "/sjr/beodata/local/python_linux/lib/python2.6/posixpath.py", line 52, in isabs
    return s.startswith('/')
AttributeError: StringList instance has no attribute 'startswith'
The full traceback has been saved in /tmp/sphinx-err-msDf2i.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
Either send bugs to the mailing list at <http://groups.google.com/group/sphinx-dev/>,
or report them in the tracker at <https://bitbucket.org/birkenfeld/sphinx/issues/>. Thanks!
```

Tried both 1.1.3 and the development version (in case bug had been fixed.

Dropped into the debugger and printed out 's':

```
StringList([u'__array_finalize__(self)', u'', u'This method is called whenever the system internally allocates a', u'new array from *obj*, where *obj* is a subclass (subtype) of the', u':class:`ndarray`. It can be used to change attributes of *self*', u'after construction (so as to ensure a 2-d matrix for example), or', u'to update meta-information from the "parent." Subclasses inherit', u'a default implementation of this method that does nothing.'], items=[('/sjr/beodata/programming/pydocweb/modules/numpy/doc/source/reference/arrays.classes.rst', 40), ('/sjr/beodata/programming/pydocweb/modules/numpy/doc/source/reference/arrays.classes.rst', 41), ('/sjr/beodata/programming/pydocweb/modules/numpy/doc/source/reference/arrays.classes.rst', 42), ('/sjr/beodata/programming/pydocweb/modules/numpy/doc/source/reference/arrays.classes.rst', 43), ('/sjr/beodata/programming/pydocweb/modules/numpy/doc/source/reference/arrays.classes.rst', 44), ('/sjr/beodata/programming/pydocweb/modules/numpy/doc/source/reference/arrays.classes.rst', 45), ('/sjr/beodata/programming/pydocweb/modules/numpy/doc/source/reference/arrays.classes.rst', 46), ('/sjr/beodata/programming/pydocweb/modules/numpy/doc/source/reference/arrays.classes.rst', 47)])
```

Attached the log file and the rst file that causes the problem.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/915
- Originally reported by: Tim Cera
- Originally created at: 2012-04-28T19:04:19.918

**Comment by shimizukawa** [comment]
_From Tim Cera on 2012-05-05 01:45:18+00:00_

After looking around at the code, suspect that this error is with numpydoc or the numpy sphinx extension.  Pardon the noise.

**Comment by f0k** [comment]
For anybody searching for this error, this has been fixed with numpydoc 0.7 (released June 20th, 2017).
